protocol/validation/tx.go

```diff
@@ -48,6 +48,13 @@ func ConfirmTx(snapshot *state.Snapshot, block *bc.Block, tx *bc.Tx) error {
 			if txin.AssetVersion != 1 {
 				continue
 			}
+			if ii.InitialBlock != initialBlockHash {
+				return errors.WithDetail(ErrBadTx, "issuance is for different blockchain")
+			}
+			assetID := bc.ComputeAssetID(ii.IssuanceProgram, initialBlockHash, ii.VMVersion)
```

Comment:
Hm, doing that reveals how redundant the assetID check is (because `ii.AssetID()` simply does `ComputeAssetID` of the very same values). I've _removed_ the assetID check. Please verify this still LGTY.